x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas.ts

```diff
@@ -51,23 +56,46 @@ export const createSignalsSchema = Joi.object({
   enabled: enabled.default(true),
   false_positives: false_positives.default([]),
   filter: filter.when('type', { is: 'filter', then: Joi.required(), otherwise: Joi.forbidden() }),
-  filters: filters.when('type', { is: 'query', then: Joi.optional(), otherwise: Joi.forbidden() }),
+  filters: Joi.when('type', {
+    is: 'query',
+    then: filters.optional(),
+    otherwise: Joi.when('type', {
+      is: 'saved_query',
+      then: filters.optional(),
+      otherwise: Joi.forbidden(),
+    }),
+  }),
   from: from.required(),
   rule_id,
   immutable: immutable.default(false),
   index: index.required(),
   interval: interval.default('5m'),
-  query: query.when('type', { is: 'query', then: Joi.required(), otherwise: Joi.forbidden() }),
-  language: language.when('type', {
+  query: Joi.when('type', {
```

Comment:
the Query object looks like that from `src/plugins/data/common/query/types.ts`
export interface Query {
  query: string | { [key: string]: any };
  language: string;
}
```
I was expecting to send you the same object back `{ query : { query: '', language: 'kuery' } }` and not be flatten